domain/state.go

```diff
@@ -44,3 +50,27 @@ func (st *StateBase) DB() (database.TxnRunner, error) {
 
 	return st.db, nil
 }
+
+// Prepare prepares a SQLair query. If the query has been prepared previously it
+// is retrieved from the statement cache.
+//
+// Note that because the type samples are not considered when retrieving a query
+// from the cache, it is possible that two queries may have identical text, but
+// use different types. Retrieving the wrong query would result in an error when
+// the query was passed the wrong type at execution.
+// This issue should be relatively rare and caught by QA if present.
```

Comment:
I would mention that as the cache is for individual states, conflicts between different domains is reduced.

Reply:
Yea, good to mention that explicitly, changed.